packages/web3-eth-contract/src/types.ts

```diff
@@ -20,3 +23,72 @@ export interface ContractInitOptions {
 	readonly gasLimit: Uint;
 	readonly provider: SupportedProviders<EthExecutionAPI> | string;
 }
+
+// TODO: Add correct type
```

Comment:
for all these TODOs could you open a new issue and mention these in that, will be easier to plan in sprints.